blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/EnumMetricData.java

```diff
@@ -0,0 +1,199 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+
+/**
+ * This class uses the driver specific {@link EnumReader} to
+ * read/write enum metrics from Cassandra and convert them to
+ * {@link com.rackspacecloud.blueflood.outputs.formats.MetricData}
+ * objects that are used by services such as Rollup, Ingest and Query
+ */
+public class EnumMetricData {
+
+    private static final Logger LOG = LoggerFactory.getLogger(EnumMetricData.class);
+
+    private static final Configuration configuration = Configuration.getInstance();
+    private static final ExecutorService taskExecutor = new ThreadPoolBuilder()
+                                            .withUnboundedQueue()
+                                            .withCorePoolSize(configuration.getIntegerProperty(CoreConfig.ENUM_READ_THREADS))
+                                            .withMaxPoolSize(configuration.getIntegerProperty(CoreConfig.ENUM_READ_THREADS))
+                                            .withName("Read Enum Values")
+                                            .build();
+    private static final MetadataCache metadataCache = MetadataCache.getInstance();
+
+    private EnumReader enumReader;
+
+    /**
+     * Constructor. Takes an instance or EnumReader, which would be
+     * driver specific
+     *
+     * @param reader
+     */
+    public EnumMetricData(EnumReader reader) {
+        enumReader = reader;
+    }
+
+    /**
+     * This method fetches enum metrics for a particular range, granularity, and {@link com.rackspacecloud.blueflood.types.Locator}.
+     * The result is organized in a {@link com.rackspacecloud.blueflood.outputs.formats.MetricData} object.
+     *
+     * @param locator
+     * @param range
+     * @param gran
+     * @return
+     */
+    public MetricData getEnumMetricDataForRange(final Locator locator, final Range range, final Granularity gran) {
+        Map<Locator, MetricData> metricDataMap = getEnumMetricDataForRangeForLocatorList(new ArrayList<Locator>(){{add (locator); }}, range, gran);
+        return metricDataMap.get(locator);
+    }
+
+    /**
+     * This method fetches all enum metrics for a particular range and granularity by their {@link com.rackspacecloud.blueflood.types.Locator}.
+     * The result is organized in a map of Locator -> {@link com.rackspacecloud.blueflood.outputs.formats.MetricData}.
+     *
+     * @param locators
+     * @param range
+     * @param gran
+     * @return
+     */
+    public Map<Locator, MetricData> getEnumMetricDataForRangeForLocatorList(final List<Locator> locators, final Range range, final Granularity gran) {
+        String columnFamily = CassandraModel.getColumnFamily(BluefloodEnumRollup.class, gran).getName();
+        if ( columnFamily == null ) {
```

Comment:
:) Looking at this method `CassandraModel.getColumnFamily`, it didnt give me an impression that we will get `null` as return value.